Cli/Command/CheckViolationsCommand.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+/*
+ * This file is part of the SensioLabsInsight package.
+ *
+ * (c) Lhassan Baazzi <baazzilhassan@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace SensioLabs\Insight\Cli\Command;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Gitonomy\Git\Repository as GitRepository;
+use Gitonomy\Git\RevisionList as GitRevisionList;
+
+class CheckViolationsCommand extends Command implements NeedConfigurationInterface
+{
+    protected function configure()
+    {
+        $this
+            ->setName('check-violations')
+            ->addArgument('project-uuid', InputArgument::REQUIRED, 'SensioLabs insight project UUID')
+            ->addOption('repository'    , null, InputOption::VALUE_REQUIRED, 'The GIT repository path, default to current working directory', getcwd())
+            ->addOption('commits'       , null, InputOption::VALUE_IS_ARRAY | InputOption::VALUE_REQUIRED, 'List of commits hashes')
+            ->addOption('no-global'     , null, InputOption::VALUE_NONE, 'Disable showing global violations', null)
+            ->addOption('ignore-assets' , null, InputOption::VALUE_NONE, 'Ignore assets violations(*.css, *.js, *.coffee, *.less, *.sass)', null)
+            ->setDescription('Check for violations occurred in one or multiple commits in the last analysis.')
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $projectUuid = $input->getArgument('project-uuid');
+        $ignoreAssets = $input->getOption('ignore-assets');
+        $noGlobalViolations = $input->getOption('no-global');
+
+        $api = $this->getApplication()->getApi();
+
+        $gitRepository = new GitRepository($input->getOption('repository'));
+
+        if ($gitRepository->isBare()) {
+            throw new \Exception(
+                sprintf('No Git repository found on this path "%s"', $input->getOption('repository'))
+            );
+        }
+
+        $output->writeln(sprintf('<info>Found GIT repository in <%s></info>', $gitRepository->getGitDir()));
+
+        if ($input->getOption('commits')) {
+            $revisionsList = new GitRevisionList($gitRepository, $input->getOption('commits'));
+            $diff = $gitRepository->getDiff($revisionsList);
+
+        } else {
+            $diff = $gitRepository->getHeadCommit()->getDiff();
+        }
+
+        if (!$diff) {
+            $output->writeln('<info>No diff found.</info>');
+
+            return;
+        }
+
+        $changedFiles = array();
+        foreach ($diff->getFiles() as $file) {
+            if ($file->getNewName()) {
+                if (!$ignoreAssets && $this->isAsset($file->getNewName())) {
```

Comment:
The condition should be ``if ($ignoreAssets && $this->isAsset($file->getNewName()))``. Otherwise you are ignoring assets when being asked to not ignore them only